tests/utapi/utilities.js

```diff
@@ -507,4 +513,51 @@ describe('utapi v2 pushmetrics utility', () => {
             });
         });
     });
+
+    describe('with lifecycle enabled', () => {
+        _config.lifecycleRoleName = 'lifecycleTestRoleName';
+        const eventFilterList = new Set([
+            'getBucketAcl',
+            'getBucketCors',
+            'getBucketLocation',
+            'getBucketNotification',
+            'getBucketObjectLock',
+            'getBucketReplication',
+            'getBucketVersioning',
+            'getBucketWebsite',
+            'getObjectTagging',
+            'headObject',
+        ]);
+
+        testEvents
+        .map(event => {
+            const modifiedEvent = event;
+            const authInfo = makeAuthInfo('accesskey1', 'Bart');
+            authInfo.arn = `foo:assumed-role/${_config.lifecycleRoleName}/backbeat-lifecycle`;
+            modifiedEvent.metrics.authInfo = authInfo;
+            modifiedEvent.metrics.canonicalID = 'accesskey1';
+            return modifiedEvent;
+        })
+        .map(event => {
+            if (eventFilterList.has(event.action)) {
+                it(`should skip action ${event.action}`, () => {
+                    _config.lifecycleRoleName = 'lifecycleTestRoleName';
+                    const eventPushed = pushMetric(event.action, log, event.metrics);
+                    assert.strictEqual(eventPushed, undefined);
+                });
+            }
+            return event;
+        })
+        .forEach(event => {
+            if (!eventFilterList.has(event.action)) {
+                it(`should compute and push metrics for ${event.action}`, () => {
+                    const eventPushed = pushMetric(event.action, log, event.metrics);
+                    assert(eventPushed);
+                    Object.keys(event.expected).forEach(key => {
+                        assert.strictEqual(eventPushed[key], event.expected[key]);
+                    });
```

Comment:
Should this be `assert.strictDeepEquals`? Or is some stuff in event pushed optional? Maybe?